drizzlepac/runastrodriz.py

```diff
@@ -2133,6 +2205,52 @@ def _analyze_exposure(filename):
     return process_exposure
 
 
+def _update_idcscale(filename):
+    """
+    Update IDCSCALE if a headerlet does not include it.
+
+    Parameters
+    ----------
+    filename : str or fits.HDUList
+        The science file to be updated.
+        If HDUList it must be opened in "update" mode.
+
+    """
+
+    if isinstance(filename, str):
+        hdul = fits.open(filename, mode='update')
+    else:
+        # Assume it's an HDUList object in "update" mode.
+        hdul = filename
+    num_sci = fileutil.countExtn(hdul)
+    # get IDCTAB name
+    itabroot = hdul[0].header['idctab'].split('$')[1].split('_')[0]
+    hdu_idscale = None
```

Comment:
It does not matter, but you initialize _hdu_idscale_ here, but then it is never used.  The variable f_hdu_idscale_ is actually is use.